### PR TITLE
Do not show 'work link' if work transfer is cancelled'

### PR DESCRIPTION
**Pull request by njaffer**
related to #1552

### CODE REVIEW

File: app/helpers/sufia/sufia_helper_behavior.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@ def error_messages_for(object)
     end
 
     def show_transfer_request_title(req)
-      if req.deleted_work?
+      if req.deleted_work? || req.status != 'Canceled'
```
Comment (jcoyne) [inline app/helpers/sufia/sufia_helper_behavior.rb]: Have you considered making this a method on the `req` object?

Comment (njaffer) [inline app/helpers/sufia/sufia_helper_behavior.rb]: Sure, I will update the code

File: app/models/proxy_deposit_request.rb [hunk 2]
```diff
@@ -75,6 +75,10 @@ def cancel!
     save!
   end
 
+  def is_canceled?
+    self.status == 'canceled'
```
Comment (jcoyne) [inline app/models/proxy_deposit_request.rb]: Let's make `'canceled'` a constant (and use it in `cancel!`) too.  Also, can you add a test to this PR?

File: app/models/proxy_deposit_request.rb [hunk 3]
```diff
@@ -75,6 +75,10 @@ def cancel!
     save!
   end
 
+  def is_canceled?
```
Comment (jcoyne) [inline app/models/proxy_deposit_request.rb]: The ruby convention is just `canceled?` without having `is_` at the front.

**Comment by njaffer** [comment]
@jcoyne: The test already covers the canceled status here: https://github.com/projecthydra/sufia/blob/master/spec/models/proxy_deposit_request_spec.rb#L63

and for making 'canceled' as a constant, we will have to create all other status as constants also, how about we make that as a separate ticket?

**Comment by jcoyne** [comment]
There is no test of the behavior of the new method or how it affects the view.

**Comment by njaffer** [comment]
@jcoyne: Is there an existing test that needs to be modified? I cannot seem to find where would this test go?

**Comment by jcoyne** [comment]
@njaffer I'd put one test in here: https://github.com/njaffer/sufia/blob/transfer/spec/models/proxy_deposit_request_spec.rb  and then one in here: https://github.com/projecthydra/sufia/blob/master/spec/helpers/sufia_helper_spec.rb  I did mistype before, I said view, but I actually meant helper.

**Comment by awead** [comment]
:ok_hand: to me, but @jcoyne might want a test in `sufia_helper_spec.rb`.

**Comment by njaffer** [comment]
@awead , @jcoyne : I do not see any relevant test for testing the link for work name in sufia_helper_spec.rb 

The test I added in proxy_deposit_request_spec.rb tests the code I have added in the proxy_deposit_request file

**Comment by jcoyne** [comment]
@njaffer Do you think it might be important to test the behavior of show_transfer_request_title when the status of the request is canceled? If you don't add a test here your change might be reverted without anyone knowing this was important behavior.

**Comment by njaffer** [comment]
@jcoyne: added the test to make sure no one can put title as a link when the transferred work is cancelled!

**Pull request merged**